rotkehlchen/globaldb/upgrades/v5_v6.py

```diff
@@ -0,0 +1,63 @@
+import logging
+from typing import TYPE_CHECKING
+
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import GeneralCacheType
+
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.drivers.gevent import DBConnection, DBCursor
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+V5_V6_UPGRADE_UNIQUE_CACHE_KEYS: set[GeneralCacheType] = {
+    GeneralCacheType.CURVE_POOL_ADDRESS,
+    GeneralCacheType.MAKERDAO_VAULT_ILK,
+    GeneralCacheType.CURVE_GAUGE_ADDRESS,
+}
+
+
+def _create_new_tables(cursor: 'DBCursor') -> None:
+    log.debug('Enter _create_new_tables')
+    cursor.execute(
+        """
+        CREATE TABLE IF NOT EXISTS unique_cache (
+            key TEXT NOT NULL PRIMARY KEY,
+            value TEXT NOT NULL,
+            last_queried_ts INTEGER NOT NULL
+        );
+        """,
+    )
+    log.debug('Exit _create_new_tables')
+
+
+def _transfer_unique_cache_data(cursor: 'DBCursor') -> None:
+    log.debug('Enter _transfer_unique_cache_data')
+    table_exists = cursor.execute(
+        'SELECT COUNT(*) FROM sqlite_master '
+        'WHERE type="table" AND name="general_cache"',
+    ).fetchone()[0] == 1
+    if table_exists:
+        for key_part in V5_V6_UPGRADE_UNIQUE_CACHE_KEYS:
```

Comment:
I had another though. What happens here if some of the `V5_V6_UPGRADE_UNIQUE_CACHE_KEYS` have multiple values? What does this code end up doing? Does it just insert the first one and ignores the rest?

Can you make this also part of the DB upgrade test you have made?

Reply:
Oww I see.

Alright. But what would you want it to do in that case?

Reply:
okayy

Reply:
In other to get the skipped values before logging them could be a little expensive because, we would have to do the query in 3 parts.
1. `Select` first from general using incomplete key part.
2. `Select` again with complete key gotten from first `Select` if value already exists in unique_cache and if exists, keep them in an array to log them later.
3. then Save to unique_cache if not already in unique_cache.

Unless there is a shorter way to proceed.

Reply:
hmm perhaps it's not important then. Forget about this.